src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -21,13 +26,27 @@
     public static final byte T_FOLLOWED_BY_INSERTION_BASE = (byte) 89;
     public static final byte G_FOLLOWED_BY_INSERTION_BASE = (byte) 90;
 
-    protected final GATKRead read;         // the read this base belongs to
-    protected final int offset;                 // the offset in the bases array for this base
+    final GATKRead read;         // the read this base belongs to
+    final int offset;            // the offset in the bases array for this base
 
     private final CigarElement currentCigarElement;
     private final int currentCigarOffset;
     private final int offsetInCurrentCigar;
 
+    public static final Comparator<PileupElement> COMPARATOR = (p1, p2) -> {
+        if (p1.offset < p2.offset) {
+            return -1;
+        } else if (p1.offset > p2.offset) {
+            return 1;
+        } else if (p1.read.getStart() < p2.read.getStart()) {
+            return -1;
+        } else if (p1.read.getStart() > p2.read.getStart()) {
+            return 1;
+        } else {
+            return 0;
```

Comment:
The PileupElements are equal when offset and start are the same? What if they are different bases?